### ISSUE TITLE
WithAttr missing from the GraphDefBuilder Options

### ISSUE THREAD

**Issue by sinjax** [bytedeco/javacpp-presets#160]
I was testing javacpp's tensorflow preset by implementing the MNIST classifier demo from python.
Stuck at the first hurdle im afraid. The demo requires the ability to set a placeholder variable. This in turn requires the ability to set the shape of the Placeholder using a TensorShape by calling WithAttr.

The offending lines in the original c++ can be found in graph_def_builder.h and the functions missing from tensorflow.java are these. 

Problem being that tensorflow seems to use this WithAttr stuff _very_ liberally indeed. A lot of the settings of graph nodes seem to be controlled with it. Any thoughts on this?

```
    // Override the default value for an optional attr.
    template <class T>
    Options WithAttr(StringPiece attr_name, T&& value) const {
      return Options(*this).WithAttrImpl(attr_name, std::forward<T>(value));
    }
    // Note: overload needed to allow {...} expressions for value.
    template <class T>
    Options WithAttr(StringPiece attr_name,
                     std::initializer_list<T> value) const {
      return WithAttr<std::initializer_list<T>>(attr_name, std::move(value));
    }
```

**Comment by saudet** [comment]
Very liberally... would you have a list of concrete types this is used with?

**Comment by sinjax** [comment]
Managed to get this list:
```
DataType):
DataTypeSlice):
StringPiece):
TensorShape):
bool):
const
float):
gtl::ArraySlice<TensorShape>):
gtl::ArraySlice<float>):
gtl::ArraySlice<int>):
int64):
```

by running this command:
```
find . -type f -follow | grep -E 'h$|cc$' | xargs -I% grep '//.*WithAttr' % | cut -d, -f 2 | cut -d' ' -f 2 | sort | uniq
```

**Comment by saudet** [comment]
I had to update JavaCPP a bit, but the commits above should fix this! `SetAttrValue()` gets called internally, so I've taken the list of supported data types from there. Although not exactly everything fits, it should cover pretty much all that is of interest. Let me know if there's anything missing from these additions. Thanks for reporting this!

**Comment by sinjax** [comment]
niiiiice :) Trying it out now!

**Comment by sinjax** [comment]
This might be another issue but the latest tensorflow on master gives me this error:

'''
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=1G; support was removed in 8.0
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.bytedeco.javacpp-presets:tensorflow:jar:master-1.2-SNAPSHOT
[WARNING] 'version' contains an expression but should be a constant. @ org.bytedeco.javacpp-presets:tensorflow:${tensorflow.version}-${project.parent.version}, /Users/ss/Development/javacpp-presets/tensorflow/pom.xml, line 14, column 12
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] Inspecting build with total of 1 modules...
[INFO] Not installing Nexus Staging features:
[INFO]  * Preexisting staging related goal bindings found in 1 modules.
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building JavaCPP Presets for TensorFlow master-1.2-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.7:resources (generate-sources) @ tensorflow ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.3:compile (generate-sources) @ tensorflow ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- javacpp:1.2-SNAPSHOT:build (generate-sources) @ tensorflow ---
[INFO] Targeting /Users/ss/Development/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/platform/default/integral_types.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/numeric_types.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/platform/init_main.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/platform/types.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/platform/mutex.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/platform/macros.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/util/port.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/lib/core/error_codes.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/platform/logging.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/lib/core/status.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/platform/protobuf.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/platform/env.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/config.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/versions.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/public/session_options.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/lib/core/threadpool.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/allocation_description.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/allocator.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/tensor_shape.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/types.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/tensor.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/tensor_description.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/tensor_types.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/tensor_shape.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/tensor.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/attr_value.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/op_def.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/function.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/core/framework/graph.pb.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/public/session.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/public/tensor_c_api.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/op_def_builder.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/op_def_util.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/op.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/types.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/graph/edgeset.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/lib/gtl/iterator_range.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/graph/graph.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/graph/node_builder.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/graph/graph_def_builder.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/graph/default_device.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/cc/ops/standard_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/cc/ops/const_op.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/cc/ops/cc_op_gen.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/array_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/attention_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/data_flow_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/image_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/io_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/linalg_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/logging_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/math_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/nn_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/parsing_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/random_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/sparse_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/state_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/string_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/summary_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/training_ops.h
[INFO] Parsing /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/tensorflow/cc/ops/user_ops.h
[INFO] 
[INFO] --- maven-dependency-plugin:2.10:copy-dependencies (copy-dependencies) @ tensorflow ---
[INFO] Copying javacpp-1.2-SNAPSHOT.jar to /Users/ss/Development/javacpp-presets/target/javacpp.jar
[INFO] org.bytedeco.javacpp-presets:tensorflow:jar:macosx-x86_64:master-1.2-SNAPSHOT already exists in destination.
[INFO] 
[INFO] --- maven-resources-plugin:2.7:resources (default-resources) @ tensorflow ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.3:compile (default-compile) @ tensorflow ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /Users/ss/Development/javacpp-presets/tensorflow/target/classes
[INFO] /Users/ss/Development/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/helper/tensorflow.java: Some input files use unchecked or unsafe operations.
[INFO] /Users/ss/Development/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/helper/tensorflow.java: Recompile with -Xlint:unchecked for details.
[INFO] 
[INFO] --- javacpp:1.2-SNAPSHOT:build (process-classes) @ tensorflow ---
[INFO] Generating /Users/ss/Development/javacpp-presets/tensorflow/target/classes/org/bytedeco/javacpp/jnitensorflow.cpp
[INFO] Compiling /Users/ss/Development/javacpp-presets/tensorflow/target/classes/org/bytedeco/javacpp/macosx-x86_64/libjnitensorflow.dylib
[INFO] clang++ -I/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/ -I/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-genfiles/ -I/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-tensorflow-master/ -I/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/google/protobuf/src/ -I/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/third_party/eigen3/ -I/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-tensorflow-master/external/eigen_archive/ -I/Library/Java/JavaVirtualMachines/jdk1.8.0_73.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk1.8.0_73.jdk/Contents/Home/include/darwin /Users/ss/Development/javacpp-presets/tensorflow/target/classes/org/bytedeco/javacpp/jnitensorflow.cpp -std=c++11 -Wl,-rpath,@loader_path/. -march=x86-64 -m64 -Wall -O3 -fPIC -dynamiclib -undefined dynamic_lookup -o /Users/ss/Development/javacpp-presets/tensorflow/target/classes/org/bytedeco/javacpp/macosx-x86_64/libjnitensorflow.dylib -L/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-bin/tensorflow/cc/ -Wl,-rpath,/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-bin/tensorflow/cc/ -ltensorflow -framework JavaVM 
In file included from /Users/ss/Development/javacpp-presets/tensorflow/target/classes/org/bytedeco/javacpp/jnitensorflow.cpp:81:
In file included from /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/allocator.h:25:
In file included from /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/type_traits.h:22:
In file included from /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/tensorflow/core/framework/types.h:23:
In file included from /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from /Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-tensorflow-master/external/eigen_archive/eigen-eigen-ed4c9730b545/unsupported/Eigen/CXX11/Tensor:14:
/Users/ss/Development/javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-master/bazel-tensorflow-master/external/eigen_archive/eigen-eigen-ed4c9730b545/unsupported/Eigen/CXX11/Core:16:10: fatal error: 
      'Eigen/src/Core/util/DisableStupidWarnings.h' file not found
#include <Eigen/src/Core/util/DisableStupidWarnings.h>
         ^
1 error generated.
'''

**Comment by sinjax** [comment]
Ah found the problem! different location for eigen :)

**Comment by saudet** [comment]
Yeah, I don't understand we have to hard code that there, but that's how
tensorflow does it in its own build files...

**Comment by saudet** [comment]
Included in the just released version 1.2! Enjoy
